jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -88,6 +89,115 @@ public HttpFields(HttpFields fields)
+     * </pre>
+     *
+     * <p>This method can be used to coalesce many fields into one:</p>
+     * <pre>
+     * // Input:
+     * GET / HTTP/1.1
+     * Host: localhost
+     * Cookie: foo=1
+     * Cookie: bar=2,baz=3
+     * User-Agent: Jetty
+     *
+     * // Computation:
+     * httpFields.computeField("Cookie", (name, fields) ->
+     * {
+     *     // No cookies, nothing to do.
+     *     if (fields == null)
+     *         return null;
+     *
+     *     // Coalesces all cookies.
+     *     String coalesced = fields.stream()
+     *         .flatMap(field -> Stream.of(field.getValues()))
+     *         .collect(Collectors.joining(", "));
+     *
+     *     // Returns a single Cookie header with all cookies.
+     *     return new HttpField(name, coalesced);
+     * }
+     *
+     * // Output:
+     * GET / HTTP/1.1
+     * Host: localhost
+     * Cookie: foo=1, bar=2, baz=3
+     * User-Agent: Jetty
+     * </pre>
+     *
+     * <p>This method can be used to replace a field:</p>
+     * <pre>
+     * httpFields.computeField("X-Length", (name, fields) ->
+     * {
+     *     if (fields == null)
+     *         return null;
+     *
+     *     // Get any value among the X-Length headers.
+     *     String length = fields.stream()
+     *         .map(HttpField::getValue)
+     *         .findAny()
+     *         .orElse("0");
+     *
+     *     // Replace X-Length headers with X-Capacity header.
+     *     return new HttpField("X-Capacity", length);
+     * });
+     * </pre>
+     *
+     * <p>This method can be used to remove a field:</p>
+     * <pre>
+     * httpFields.computeField("Connection", (name, fields) -> null);
+     * </pre>
+     *
+     * @param name the HTTP header name
+     * @param computeFn the compute function
+     */
+    public void computeField(String name, BiFunction<String, List<HttpField>, HttpField> computeFn)
+    {
+        boolean found = false;
+        ListIterator<HttpField> iterator = listIterator();
+        while (iterator.hasNext())
+        {
+            HttpField field = iterator.next();
+            if (field.getName().equalsIgnoreCase(name))
+            {
+                if (found)
+                {
+                    // Remove other headers with the same name, since
+                    // we have computed one from all of them already.
+                    iterator.remove();
+                }
+                else
+                {
+                    found = true;
+                    HttpField newField = computeFn.apply(name, Collections.unmodifiableList(getFields(name)));
+                    if (newField == null)
+                        iterator.remove();
+                    else
+                        iterator.set(newField);
+                }
+            }
+        }
+        if (!found)
+        {
+            HttpField newField = computeFn.apply(name, null);
+            if (newField != null)
+                put(newField);
```

Comment:
working on a better impl... meanwhile use add here not put, as you have already iterated and discovered that it is not found.